go.mod

```diff
@@ -1,27 +1,31 @@
 module github.com/cli/go-gh/v2
 
-go 1.21
+go 1.23.0
+
+toolchain go1.23.7
```

Comment:
Is this really required? Would think it is unusual to have a toolchain entry for a library.

Reply:
Great question; `go mod` did this in upgrading dependencies 🤦

Reply:
Removing for now